chalk-engine/src/logic.rs

```diff
@@ -440,22 +367,9 @@ impl<C: Context> Forest<C> {
     ) {
         let mut tables = vec![];
 
-        let num_universes = C::num_universes(&self.tables[table].table_goal);
-        for canonical_strand in self.tables[table].strands_mut() {
-            // FIXME if CanonicalExClause were not held abstract, we
-            // could do this in place like we used to (and
-            // `instantiate_strand` could take ownership), since we
-            // don't really need to instantiate here to do this
-            // operation.
-            let (delayed_strand, subgoal_table) = Self::with_instantiated_strand(
-                context,
-                num_universes,
-                canonical_strand,
-                DelayStrandAfterCycle { table },
-            );
-
-            *canonical_strand = delayed_strand;
-
+        for strand in self.tables[table].strands_mut() {
+            // FIXME: do we need to uncanonicalize this
```

Comment:
I don't think so -- but I have to review this logic -- in any case I actually want to remove this "delaying" stuff, see #250, so maybe that too would be a good step to take before doing the rest of this PR, just to simplify things?